packages/expo-gl/src/Canvas.tsx

```diff
@@ -37,7 +37,18 @@ const CanvasWrapper: React.FunctionComponent<
   const ref = React.useRef<View>(null);
   const _canvasRef = React.useRef<HTMLCanvasElement>(null);
 
-  function updateCanvasSize(): void {
+  const getSize = React.useCallback((): { width: number; height: number } => {
+    if (size) {
+      return size;
+    } else if (!ref.current || typeof window === 'undefined') {
+      return { width: 0, height: 0 };
+    }
+    const element = getElement(ref.current);
+    const { offsetWidth: width = 0, offsetHeight: height = 0 } = element;
+    return { width, height };
+  }, [size?.width, size?.height])
```

Comment:
It would be better to move it to a top level and pass needed arguments as parameters (size and ref.current). It will make the code more readable, now it's unclear what is the relation between `getSize` and `onLayout`.

Reply:
Indeed, we would need to pass the current size to the function, but I believe this approach will improve readability and eliminate the need to track dependencies. The `onLayout` function must remain within the component since it modifies the current state. I was only referring to the `getSize` function.

Reply:
> Indeed, we would need to pass the current size to the function, but I believe this approach will improve readability and eliminate the need to track dependencies. The `onLayout` function must remain within the component since it modifies the current state. I was only referring to the `getSize` function.

Hi, I needed to create a new PR (https://github.com/expo/expo/pull/36214) because of ongoing merge conflict issues. Please use this one going forward. Thanks!